Doppelganger/iOS/UIKit+Doppelganger.m

```diff
@@ -76,10 +76,42 @@ - (void)wml_applyBatchChanges:(NSArray *)changes inSection:(NSUInteger)section w
     [self beginUpdates];
     [self deleteRowsAtIndexPaths:deletion withRowAnimation:animation];
     [self insertRowsAtIndexPaths:insertion withRowAnimation:animation];
-        for (WMLArrayDiff *diff in moving) {
-            [self moveRowAtIndexPath:[NSIndexPath indexPathForItem:diff.previousIndex inSection:section]
-                         toIndexPath:[NSIndexPath indexPathForItem:diff.currentIndex inSection:section]];
+    for (WMLArrayDiff *diff in moving) {
+        [self moveRowAtIndexPath:[NSIndexPath indexPathForItem:diff.previousIndex inSection:section]
+                     toIndexPath:[NSIndexPath indexPathForItem:diff.currentIndex inSection:section]];
+    }
+    [CATransaction setCompletionBlock:completion];
```

Comment:
I understand the goal behind it, but I don't think this is a great way of telling whether table view has finished reloading,
Why don't you do it on the consumer side where you calling `wml_applyBatchChanges:inSection:withRowAnimation:`?